packages/wrangler/src/dev.ts

```diff
@@ -626,37 +625,41 @@ export async function startDev(args: StartDevOptions) {
 		} else {
 			devEnv = new DevEnv();
 
-			// The ProxyWorker will have a stable host and port, so only listen for the first update
-			void devEnv.proxy.ready.promise.then(({ url }) => {
-				if (args.onReady) {
-					args.onReady(url.hostname, parseInt(url.port));
-				}
-
-				if (
-					(args.enableIpc || !args.onReady) &&
-					process.send &&
-					typeof vitest === "undefined"
-				) {
-					process.send(
-						JSON.stringify({
-							event: "DEV_SERVER_READY",
-							ip: url.hostname,
-							port: parseInt(url.port),
-						})
-					);
-				}
-			});
-
 			await setupDevEnv(devEnv, args.config, authHook, args);
 
 			if (isInteractive() && args.showInteractiveDevSession !== false) {
 				unregisterHotKeys = registerDevHotKeys([devEnv], args);
 			}
 		}
 
+		const [primaryDevEnv, ...secondary] = Array.isArray(devEnv)
+			? devEnv
+			: [devEnv];
+
+		// The ProxyWorker will have a stable host and port, so only listen for the first update
+		void primaryDevEnv.proxy.ready.promise.then(({ url }) => {
+			if (args.onReady) {
+				args.onReady(url.hostname, parseInt(url.port));
+			}
+
+			if (
+				(args.enableIpc || !args.onReady) &&
+				process.send &&
+				typeof vitest === "undefined"
+			) {
+				process.send(
+					JSON.stringify({
+						event: "DEV_SERVER_READY",
+						ip: url.hostname,
+						port: parseInt(url.port),
+					})
+				);
+			}
+		});
```

Comment:
I moved this out so that `runWranglerDev` works can wait for the IPC message properly and providing multiple configs.